### DISCUSSION TITLE
Doc's for adding new Language LSP for zed

### DISCUSSION THREAD

**Discussion by SPR-Ideas** [zed-industries/zed#6662] [Feature Requests]
Hey Zed Community , 
i am using Zed for past 2 months , I really excited with weightiness of Zed , tat's mind blowing, One thing i found could be improved is , i cannot customise the LSP for instance , I have Bicep Lsp but i could integrate it with zed 🙁 i don't know why the reason is so , can any one help me with that pls ?

**Comment by vasco3** [comment]
I would like to add ReScript language

**Comment by nathansbradshaw** [comment]
Now that it is open source you can look at the code and figure out how to add a LSP, but there are some things that look to be required. Your desired language needs a `tree-sitter` as a cargo crate. It also needs a LSP. You add in a <language_name>.rs file into the languages lib, then you load that language file into the `languages.rs` file. you also need to define some configurations in a folder called `<language> in the languages lib. That's as far as I've gotten. I think at that point you just need to do some testing and tweaking.

It should be noted, however, that the goal of Zed isn't to have people add LSPs as they please, but instead to build a plugin manager that you can then load in LSPs as needed. So If you want to add the LSP in the way desired by the developers, then helping with the plugin manager is the correct move forward.

**Reply by vasco3** [reply]
ReScript does have tree-sitter
https://github.com/nkrkv/tree-sitter-rescript
and a Language Server https://github.com/rescript-lang/rescript-vscode/tree/master/server

so far so good

**Reply by nathansbradshaw** [reply]
There is a PR open for Zig support that would be another good resource for learning how to add lsps
Link: https://github.com/zed-industries/zed/pull/6690

**Comment by codeslubber** [comment]
Is this how support for Dioxus would have to be added?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
